implementations/rust/ockam/ockam_identity/src/identities/storage/change_history_repository_sql.rs

```diff
@@ -35,15 +37,50 @@ impl ChangeHistorySqlxDatabase {
 
 #[async_trait]
 impl ChangeHistoryRepository for ChangeHistorySqlxDatabase {
+    async fn update_identity(&self, identity: &Identity) -> Result<()> {
+        let mut transaction = self.database.begin().await.into_core()?;
+        let query1 = query_as("SELECT * FROM identity WHERE identifier=$1")
+            .bind(identity.identifier().to_sql());
+        let row: Option<ChangeHistoryRow> =
+            query1.fetch_optional(&mut *transaction).await.into_core()?;
+
+        let do_insert = match row {
+            Some(row) => {
+                let known_identity = Identity::import_from_change_history(
+                    Some(identity.identifier()),
+                    row.change_history()?,
+                    Vault::create_verifying_vault(),
```

Comment:
Do you think it should be injected dependency? Not a huge difference and I don't think we're going to have a different VerifyingVault, but still